pkg/backup/backupresolver/targets.go

```diff
@@ -674,35 +676,472 @@ func LoadAllDescs(
+				addDesc(schema)
+			}
+			return nil
+		}
+
+		// Helper to check if a schema is virtual/system (should be skipped).
+		isVirtualSchema := func(schema catalog.SchemaDescriptor) bool {
+			if schema == nil {
+				return false
+			}
+			return schema.SchemaKind() == catalog.SchemaVirtual
+		}
+
+		// Process DATABASE targets.
+		for _, dbName := range targets.Databases {
+			db, err := col.ByName(txn.KV()).Get().Database(ctx, string(dbName))
+			if err != nil {
+				return err
+			}
+			if !db.Public() {
+				return sqlerrors.NewUndefinedDatabaseError(tree.ErrString(&dbName))
+			}
+			addDesc(db)
+
+			// Track completeDBs and requestedDBs for DATABASE targets.
+			expandedDBs = append(expandedDBs, db.GetID())
+			requestedDBs = append(requestedDBs, db)
+
+			// For database backup, get all objects in the database.
+			allInDB, err := col.GetAllInDatabase(ctx, txn.KV(), db)
+			if err != nil {
+				return err
+			}
+
+			err = allInDB.ForEachDescriptor(func(desc catalog.Descriptor) error {
+				// Skip virtual tables and temp tables.
+				if table, ok := desc.(catalog.TableDescriptor); ok {
+					if desc.Dropped() || (!desc.Public() && !desc.Offline()) {
+						return nil
+					}
+					if table.IsVirtualTable() || table.IsTemporary() {
+						return nil
+					}
+					if err := getReferencedTypes(table, db); err != nil {
+						return err
+					}
+					if err := getReferencedFunctions(table); err != nil {
+						return err
+					}
+					if err := addSchemaDesc(table.GetParentSchemaID()); err != nil {
+						return err
+					}
+					addDesc(table)
+					return nil
+				}
+
+				// For schemas, only skip if dropped (include PUBLIC, OFFLINE, and ADD).
+				// Skip virtual and temporary schemas.
+				if schema, ok := desc.(catalog.SchemaDescriptor); ok {
+					if schema.Dropped() {
+						return nil
+					}
+					if isVirtualSchema(schema) || schema.SchemaKind() == catalog.SchemaTemporary {
+						return nil
+					}
+					addDesc(schema)
+					return nil
+				}
+
+				// For other descriptors (types, functions), apply standard filter.
+				if desc.Dropped() || (!desc.Public() && !desc.Offline()) {
+					return nil
+				}
+				addDesc(desc)
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		// Process TABLE targets.
+		for _, origPattern := range targets.Tables.TablePatterns {
+			pattern, err := origPattern.NormalizeTablePattern()
+			if err != nil {
+				return err
+			}
+
+			switch p := pattern.(type) {
+			case *tree.TableName:
+				// Resolve individual table.
+				un := p.ToUnresolvedObjectName()
+
+				found, prefix, desc, err := resolver.ResolveExisting(
+					ctx, un, r, tree.ObjectLookupFlags{DesiredObjectKind: tree.TableObject}, currentDatabase, searchPath,
+				)
+				if err != nil {
+					return err
+				}
+				if !found {
+					return errors.Errorf("table %q does not exist", tree.ErrString(p))
+				}
+
+				table, ok := desc.(catalog.TableDescriptor)
+				if !ok || !table.Public() || table.IsTemporary() {
```

Comment:
nit: if `!ok` is returned, returning does not exist makes sense. But for non public or temp tables, perhaps the error should be `cannot be backed up`?